### PR TITLE
Pass arg lists to gcloud as comma-separated single arg

### LINKED ISSUE #8117: Scary warnings spinning up e2e on GCE
This happens over and over.  Scary messages.

WARNING: Lists should be separated by commas, try "--allow=tcp,udp,icmp,esp,ah,sctp". If you intend to use udp, icmp, esp, ah, sctp for positional arguments, put them before the flag.
WARNING: Lists should be separated by commas, try "--scopes=storage-ro,compute-rw". If you intend to use compute-rw for positional arguments, put them before the flag.
WARNING: Lists should be separated by commas, try "--metadata-from-file=startup-script=/usr/local/google/home/thockin/src/kubernetes/hack/e2e-internal/../../cluster/../cluster/gce/../../cluster/../cluster/gce/configure-vm.sh,kube-env=/tmp/kubernetes.MURmzX/master-kube-env.yaml". If you intend to use kube-env=/tmp/kubernetes.MURmzX/master-kube-env.yaml for positional arguments, put them before the flag.
WARNING: Lists should be separated by commas, try "--disk=name=e2e-test-thockin-master-pd,device-name=master-pd,mode=rw,boot=no,auto-delete=no". If you intend to use device-name=master-pd, mode=rw, boot=no, auto-delete=no for positional arguments, put them before the flag.

### PR DESCRIPTION
**Pull request by justinsb**
This should get rid of the scary warning messages on GCE up.

Fixes #8117

### CODE REVIEW

**Comment by justinsb** [comment]
Sadly, this error still remains:

```
WARNING: You have selected a disk size of under [200GB]. This may result in poor I/O performance. For more information, see: https://developers.google.com/compute/docs/disks/persistent-disks#pdperformance.
```

Pretty easy fix, but it's a behavioural change.

**Comment by roberthbailey** [comment]
@zmerlynn FYI.

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -456,10 +456,13 @@ function add-instance-metadata-from-file {
   local attempt=0
   while true; do
     echo "${kvs[@]}"
+    # Join using commas as delimiter:
+    local joined_kvs=$( IFS=$','; echo "${kvs[*]}" )
```
Comment (lavalamp) [inline cluster/gce/util.sh]: suggest making a function to do this for clarity, even though it's a one liner.

**Comment by piosz** [comment]
cc @fgrzadkowski

**Comment by fgrzadkowski** [comment]
@justinsb Can you please rebase? I've started working on this, but then @piosz pointed me to this PR. Let's move it forward.

**Comment by justinsb** [comment]
It'll be a few hours until I can get to this.  Is that OK?

**Comment by justinsb** [comment]
Rebased!  I noticed that some of this had been done concurrently, with the join function effectively inlined.  lavalamp suggested pulling it out as a real function.  I have it as a separate line.  Any opinions?

**Comment by jlowdermilk** [comment]
+1 to pulling it out as a function.

**Comment by justinsb** [comment]
Created a function join_csv

**Comment by jlowdermilk** [comment]
LGTM

**Pull request merged**